keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -711,214 +647,324 @@ BucketAccess {
+    // +required
     BucketAccessClassName string
 
-    // CredentialsSecretName is the name of the secret that COSI should populate
-    // with the credentials. If a secret by this name already exists, then it is
-    // assumed that credentials have already been generated. It is not overridden.
-    // This secret is deleted when the BucketAccess is delted.
-    CredentialsSecretName string
-    
-    // ServiceAccountName is the name of the serviceAccount that COSI will map
-    // to the OSP service account when IAM styled authentication is specified
-	// +optional
-	ServiceAccountName string
+    // Protocol is the name of the Protocol that this access credential is expected to support.
+    // +required
+    Protocol Protocol
+
+    // AccessSecretName is the name of the Kubernetes secret that COSI should populate with access
+    // details and credentials.
+    // This secret is deleted when the BucketAccess is deleted.
+    // +required
+    AccessSecretName string
   }
 
   Status BucketAccessStatus {
-    // AccessGranted indicates the successful grant of privileges to access the bucket
-    AccessGranted bool
-    
-    // AccountID is the unique ID for the account in the OSP. It will be populated
-    // by the COSI sidecar once access has been successfully granted.
-    // +optional
+    // ReadyToUse indicates the successful grant of privileges to access the bucket.
+    ReadyToUse bool
+
+    // AccountID is the unique ID for the account in the OSP. It will be populated by the COSI
+    // sidecar once access has been successfully granted.
     AccountID string
+
+    // AccessedBucketName is the name of the Bucket resource that the access corresponds to. This is
+    // filled in by the Controller based on the BucketClaim so that the Sidecar knows what Bucket
+    // to allow access to for this BucketAccess.
+    // TODO: will have to update this to a list when 1-access:many-buckets support is added
+    AccessedBucketName string
+
+    // DriverName holds a copy of the BucketAccessClass driver name at the time of BucketAccess
+    // provisioning. This is kept to ensure the BucketAccess can be modified/deleted even after
+    // BucketAccessClass mutation/deletion.
+    DriverName string
+
+    // Parameters holds a copy of the BucketAccessClass opaque parameters at the time of
+    // BucketAccess provisioning. These parameters are kept to ensure the BucketAccess can be
+    // modified/deleted even after BucketAccessClass mutation/deletion.
+    Parameters map[string]string
+
+    // Error holds the most recent error message, with a timestamp.
+    // This is cleared when provisioning is successful.
+    Error *TimestampedError
   }
 ```
 
-## BucketAccessClass
+The `credentialsSecretName` is the name of the Kubernetes Secret that COSI will generate containing endpoint, credentials, and other information needed to access the OSP bucket. The same Secret can be referenced by Pods to access the OSP bucket.
```

Comment:
Is it worth adding `status.secretRef` (using e.g. `LocalObjectReference`) or `status.secretSelector` so the secret can be easily retrieved programmatically?

Reply:
I think this is a good question. It also raises some more questions for me, so let's discuss.

If we never expect to output to anything other than a Secret, I don't see a lot of difference between `spec.accessSecretName` (current) and `spec.accessSecret.name` (using local obj ref). The secret should always be local, so we won't need more than the `name` field overall. 

However, I could imagine a future where we have other options for output that aren't into a local Secret. In that light, having a local secret ref, with other context could make sense. Like below maybe

```
accessInfoOutput:
  secret: // could be optional in the future
    name: <blah>
  // futureOutputOption: {}
```

Thoughts?

Reply:
Hmm, after a second thought, I might have tried to over-engineer it here. Let's keep the `credentialsSecretName` as string!